### PR TITLE
Explore stratifications

### PR DESCRIPTION
**Pull request by PierreBoyeau**

### CODE REVIEW

File: src/scvi_v2/_tree_utils.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return np.array([rgb2hex(plt.cm.viridis(i)) for i in x])
+
+    dtypes = metadata.dtypes
+    colors_mapper = {}
+    for col in metadata.columns:
+        if dtypes[col] == "object":
+            cats = metadata[col].astype("category").cat.codes
+            colors = _get_colors_from_categorical(cats)
+        elif dtypes[col] == "category":
+            colors = _get_colors_from_continuous(metadata[col])
```
Comment (justjhong) [inline src/scvi_v2/_tree_utils.py]: shoudl this be categorical too?

File: src/scvi_v2/_model.py [hunk 2]
```diff
@@ -723,3 +730,64 @@ def get_hs(inputs, cf_sample):
...
+        sample_metadata :
+            Metadata keys to plot, by default None
+        """
+        if figure_dir is not None:
+            os.makedirs(figure_dir, exist_ok=True)
+
+        # Convert metadata to hex colors
+        colors = None
+        if sample_metadata is not None:
+            sample_metadata = [sample_metadata] if isinstance(sample_metadata, str) else sample_metadata
```
Comment (justjhong) [inline src/scvi_v2/_model.py:766]: this is great, really like the automatic coloring

File: src/scvi_v2/_model.py [hunk 3]
```diff
@@ -723,3 +730,64 @@ def get_hs(inputs, cf_sample):
...
+                linkage_method=linkage_method,
+            )
+            assert dist.ndim == 2
+
+            fig = sns.clustermap(dist.to_pandas(), row_linkage=dendrogram, col_linkage=dendrogram, row_colors=colors)
+            fig.fig.suptitle(celltype_name)
+            if figure_dir is not None:
+                fig.savefig(os.path.join(figure_dir, f"{celltype_name}.png"))
+            if show_figures:
+                plt.show()
```
Comment (justjhong) [inline src/scvi_v2/_model.py:791]: might want to do plt.clf() after plt.show to clear the plot for multiple keys

**Review by justjhong — COMMENTED** [review]
look like great progress!

File: src/scvi_v2/_tree_utils.py (hunk 1 above)
Comment (PierreBoyeau) [reply src/scvi_v2/_tree_utils.py]: yes, thanks so much!

**Pull request merged**